amethyst_core/src/thread_pool.rs

```diff
@@ -0,0 +1,53 @@
+use amethyst_error::Error;
+#[cfg(not(no_threading))]
+use std::sync::Arc;
+
+#[cfg(not(no_threading))]
+#[derive(Debug, Clone)]
+pub struct ThreadPool {
+    pool: Arc<rayon::ThreadPool>,
+}
+
+#[cfg(no_threading)]
+#[derive(Debug, Clone)]
+pub struct ThreadPool {}
+
+impl ThreadPool {
+    #[cfg(not(no_threading))]
+    pub fn new(thread_count: Option<usize>) -> Result<Self, Error> {
```

Comment:
Taking an `Option` may be awkward. Two constructors would complicate things a little bit for the caller. An enum would be another option to be more explicit.